acceptance_modelisation/base_acceptance_map_creator.py

```diff
@@ -663,9 +674,32 @@ def create_acceptance_map_cos_zenith_binned(self,
         if len(cos_zenith_model) <= 1:
             logger.warning('Only one zenith bin, zenith binning deactivated')
         for obs in observations:
-            cos_zenith_observation = np.cos(obs.get_pointing_altaz(obs.tmid).zen)
-            key_closest_model = key_model[(np.abs(cos_zenith_model - cos_zenith_observation)).argmin()]
-            acceptance_map[obs.obs_id] = dict_binned_model[key_closest_model]
+            if self.use_mini_irf_computation:
+                evaluation_time, observation_time = get_time_mini_irf(obs, self.mini_irf_time_resolution)
+
+                data_obs_all = np.zeros(tuple([len(evaluation_time), ] + list(dict_binned_model[key_model[0]].data.shape))) * dict_binned_model[key_model[0]].unit
+                for i in range(len(evaluation_time)):
+                    cos_zenith_observation = np.cos(obs.get_pointing_altaz(evaluation_time[i]).zen)
+                    key_closest_model = key_model[(np.abs(cos_zenith_model - cos_zenith_observation)).argmin()]
+                    selected_model_bin = dict_binned_model[key_closest_model]
+                    data_obs_all[i] = selected_model_bin.data * selected_model_bin.unit
+
+                data_obs = generate_irf_from_mini_irf(data_obs_all, observation_time)
+
+                if type(dict_binned_model[key_model[0]]) is Background2D:
+                    acceptance_map[obs.obs_id] = Background2D(axes=dict_binned_model[key_model[0]].axes,
+                                                              data=data_obs)
+                elif type(dict_binned_model[key_model[0]]) is Background3D:
+                    acceptance_map[obs.obs_id] = Background3D(axes=dict_binned_model[key_model[0]].axes,
+                                                              data=data_obs)
```

Comment:
I don't think the marginal performance gain that could be expected (potentially even null depending on how intelligent is the interpreter) would justify the loose in lisibility of the code. Especially as you would still need a condition switch here.

Reply:
I think it would improve the visibility of the code to have a variable called "model_axes" and "model_type".  Instead of always referring to an arbitrary dict_binned_model[key_model[0]].
I don't care about performance here.

Or even a variable BackgroundND that you would use as a class. But this would require to have a dict for kwargs with or without the orientation needed in 3D.

Reply:
The orientation is useless in 2D but supported, so not a big issue.

Reply:
Done as the first proposition.